### PR TITLE
implement all remained "panic" methods

### PR DESCRIPTION
**Pull request by shibukawa**
I didn't create test programs but I just tested via compiler. There are no syntax and type errors.

### CODE REVIEW

File: gl_opengl.go [hunk 1]
```diff
@@ -1045,7 +1070,7 @@ func Uniform4iv(dst Uniform, src []int32) {
 //
 // http://www.khronos.org/opengles/sdk/docs/man3/html/glUniform.xhtml
 func UniformMatrix2fv(dst Uniform, src []float32) {
-	panic("UniformMatrix2fv: not yet implemented")
+	gl.UniformMatrix2fv(dst.Value, int32(len(src)/4), false, &src[0])
```
Comment (dmitshur) [inline gl_opengl.go]: For consistency and readability, I'd write `int32(len(src)/(2*2))` here.

File: gl_opengl.go [hunk 2]
```diff
@@ -1082,63 +1107,64 @@ func UseProgram(p Program) {
...
+	gl.VertexAttrib3fv(uint32(dst.Value), &src[0])
 }
 
 // VertexAttrib4f writes a vec4 vertex attribute.
 //
 // http://www.khronos.org/opengles/sdk/docs/man3/html/glVertexAttrib.xhtml
 func VertexAttrib4f(dst Attrib, x, y, z, w float32) {
-	panic("VertexAttrib4f: not yet implemented")
+	gl.VertexAttrib4f(uint32(dst.Value), x, y, z, w)
+
```
Comment (dmitshur) [inline gl_opengl.go]: Extra blank line here is unneeded.

**Comment by dmitshur** [comment]
Thank you very much for the contribution here, @shibukawa!

I've looked over this PR with moderate scrutiny and did not spot any major issues. Unfortunately, I don't have the bandwidth to write sample programs for all those new endpoints and check that they're working optimally myself either.

On the one hand, this PR is a strict improvement over before, which was a panic. But I do have one concern. If new people start using some of these implemented but unverified endpoints, if there's an issue, they may not realize that the code they're relying on has never been run by anyone before. Previously, it would've been very clear as they'd get a "not yet implemented" panic, but now it will quietly run and they may spend a lot of time debugging their own code before realizing the problem is inside an unverified binding method implementation.

Since neither of us have tried the new endpoints, I would consider doing the following step for now. Instead of outright removing the `panic("MethodName: not yet implemented")` messages, consider replacing them with a:

``` Go
log.Println("MethodName: not yet tested (TODO: remove this after it's confirmed to work)")
```

As soon as someone can confirm a method works in any running program, we can remove the log statement.

People will be very motivated to remove the `log.Println` calls as they'll be very spammy (and hopefully they'll submit a PR). But they'll do that only after actually using the method in their program and seeing that it renders graphics as expected.

But I think that's a safer path to traverse. What do you think about that idea?

**Comment by dmitshur** [comment]
Also, if you agree that it's a good idea but prefer not to do it yourself, let me know and I can take care of it myself. Thanks again for contributing!

**Comment by shibukawa** [comment]
OK, I understand your idea. I will add `log.Println` and push again next week.

**Comment by dmitshur** [comment]
Great, thanks!

I suggest skipping trivial single-line implementations like these:

``` diff
 func LineWidth(width float32) {
-   panic("LineWidth: not yet implemented")
+   gl.LineWidth(width)
 }
```

Very little can go wrong there. If it compiles, it's likely fine.

But do include it on non-trivial methods that could either get the pointers wrong, or may not be implemented in the most performant way, etc., such as these:

``` diff
 func ReadPixels(dst []byte, x, y, width, height int, format, ty Enum) {
-   panic("ReadPixels: not yet implemented")
+   if ty == Enum(UNSIGNED_BYTE) {
+       c.Call("readPixels", x, y, width, height, format, ty, dst)
+   } else {
+       tmpDst := make([]float32, len(dst)/4)
+       c.Call("readPixels", x, y, width, height, format, ty, tmpDst)
+       for i, f := range tmpDst {
+           binary.LittleEndian.PutUint32(dst[i*4:], math.Float32bits(f))
+       }
+   }
 }
```

But only for methods that you've not tried yourself. If you try it and it works, then it probably doesn't need the warning. Thanks again!

**Comment by shibukawa** [comment]
I added warning messages.

File: gl_opengl.go [hunk 3]
```diff
@@ -1082,63 +1113,63 @@ func UseProgram(p Program) {
 //
 // http://www.khronos.org/opengles/sdk/docs/man3/html/glValidateProgram.xhtml
 func ValidateProgram(p Program) {
-	gl.ValidateProgram(p.Value)
+	gl.ValidateProgram(uint32(p.Value))
```
Comment (dmitshur) [inline gl_opengl.go:1116]: What's the reason for this change? Was it intended?

File: gl_webgl.go [hunk 4]
```diff
@@ -352,19 +374,29 @@ func GetString(pname Enum) string {
 }
 
 func GetTexParameterfv(dst []float32, target, pname Enum) {
-	panic("GetTexParameterfv: not yet implemented")
+	dst[0] = float32(c.Call("getTexParameter", pname).Float())
```
Comment (dmitshur) [inline gl_webgl.go:377]: I'm not sure how to correctly handle situations when `len(dst) > 1`, but I think in practice that's never going to be the case at this time.

It's just an OpenGL API quirk that's more apparent with Go's type system.

**Comment by dmitshur** [comment]
I've reviewed the changes. LGTM, merging. I've left a few minor comments that can be followed up optionally afterwards.

Thank you very much @shibukawa for this awesome contribution!
